language-plutus-core/test/Spec.hs

```diff
@@ -68,23 +68,24 @@ compareType _ _                                      = False
 compareProgram :: Eq a => Program TyName Name a -> Program TyName Name a -> Bool
 compareProgram (Program _ v t) (Program _ v' t') = v == v' && compareTerm t t'
 
+-- | A 'Program' which we compare using textual equality of names rather than alpha-equivalence.
+newtype TextualProgram a = TextualProgram { unTextualProgram :: Program TyName Name a } deriving Show
+
+instance Eq a => Eq (TextualProgram a) where
+    (TextualProgram p1) == (TextualProgram p2) = compareProgram p1 p2
```

Comment:
Can't we just use `compareProgram`?

Reply:
I don't understand the question? `tripping` needs something with an `Eq` instance, so this wraps up the program so we can give it a different `Eq` instance.